src/modules/interfaces.ts

```diff
@@ -343,6 +349,16 @@ export class MapperBase<E> {
   }
 }
 
+export class RpcBase {
+  module: ModuleInterface;
+  name: string;
+  call: (ctx: Context, ...args: string[]) => Promise<string>;
```

Comment:
This type doesn't look very good to me, tbh. Instead of returning the JSON-stringified version, wouldn't it be better to enforce that it's an array of objects that all have the same keys? (Representing that last bit in Typescript is hard, but not impossible. It could just be `Promise<any[]>` for now, and still be a bit of an improvement, though.)

Then the logic in the scheduler can have the `JSON.stringify` call, instead (so it is still done).

However, it may also make sense to change the return type to something a bit different. Right now all of the columns are parsed as `text` on the other side, but that means all of the users of these custom functions have to cast columns manually.

What if we require there to be a `types` property on this object that is an object of `columnName: postgresTypeString` that the Postgres function generation code can use to automatically generate both the casting of the columns, but an explicit return column list to make usage and documentation of the RPC on the Postgres side clearer?

Reply:
Hmm, so that the return table for this custom function would be dynamic, returning something like:

```
module_name | method_name | return_column_1 | return_column_2 | return_column_n
text        | text        | type_1          | type_2          | type_n
```

But the `iasql_rpc` table would still be saving an `output::text`?

Reply:
It may be better to explain it in a voice chat, but, no, the actual return type across the RPC boundary between the engine and postgres would remain the *same*, but the return type from the RPC function definition in the engine to the code in the scheduler is where the stronger typing would exist for the actual return columns.

*Relatedly*, we could attach extra metadata to this class in the form of `columnName: postgresTypeToCastTo` that the logic in here to *generate* the actual postgres function on the other side can use to both describe the output records better *and* automatically cast the data for the user.